packages/framework/src/shared/common.ts

```diff
@@ -47,3 +51,26 @@ export const error = (value: unknown): void => {
   // eslint-disable-next-line no-console
   console.error(value);
 };
+
+export const combineExpressions = (input: string): string => {
+  // Regular expression to split the input into literals and template variables
+  // eslint-disable-next-line prefer-named-capture-group
+  const parts = input.split(/(\$\{[^}]+\})/g).filter((x) => x);
+
+  // Initialize an array to store the processed parts of the new string
+  const resultParts: string[] = [];
+
+  // Iterate over each part and process accordingly
+  parts.forEach((part) => {
+    if (part.startsWith("${") && part.endsWith("}")) {
+      // If the part is a template variable, remove the delimiters and add to resultParts
+      resultParts.push(part.slice(2, -1));
+    } else if (part) {
+      // If the part is a literal, wrap it in single quotes and add to resultParts
+      resultParts.push(`'${part}'`);
+    }
+  });
+
+  // Combine all parts with ' + ' and return the result
+  return resultParts.length ? `\${${resultParts.join(" + ")}}` : input;
+};
```

Comment:
@evshi I think this function is more good approach than string interpolation. I understand that this is big code compare to string interpolation in one line, but this function in capable to handle all use cases of expression uses into YAML, also can check for single or multiple expression and handle in same way. also it will not change the values type that string interpolation can change.

For example, in string interpolation, if the value is in integer format, it will give back as string, but using this one, it will just return integer.

I also added all use cases example on home page and also added test cases for cover all use cases